patches/server/1032-Call-BlockRedstoneEvents-for-lecterns.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Warrior <50800980+Warriorrrr@users.noreply.github.com>
+Date: Wed, 13 Sep 2023 05:46:10 +0200
+Subject: [PATCH] Call BlockRedstoneEvents for lecterns
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/LecternBlock.java b/src/main/java/net/minecraft/world/level/block/LecternBlock.java
+index d8489fc7bbc57aa2a53026c79d2e2b2718b32912..943618c3fc72aa313fc1eb327c761e20e180a780 100644
+--- a/src/main/java/net/minecraft/world/level/block/LecternBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/LecternBlock.java
+@@ -171,6 +171,16 @@ public class LecternBlock extends BaseEntityBlock {
+     }
+ 
+     private static void changePowered(Level world, BlockPos pos, BlockState state, boolean powered) {
++        // Paper start - call BlockRedstoneEvents for lecterns
++        final boolean currentlyPowered = state.getValue(LecternBlock.POWERED);
++        if (currentlyPowered != powered) {
++            final org.bukkit.event.block.BlockRedstoneEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callRedstoneChange(world, pos, currentlyPowered ? 15 : 0, powered ? 15 : 0);
++
++            if ((powered && event.getNewCurrent() != 15) || (!powered && event.getNewCurrent() != 0)) {
```

Comment:
Could probably get rid of the somewhat ugly check here by just
```java
final int currentRedstoneLevel = state.getValue(LecternBlock.POWERED) ? 15 : 0, targetRedstoneLevel = powered ? 15 : 0;
if (currentRedstoneLevel != targetRedstoneLevel) {
    final org.bukkit.event.block.BlockRedstoneEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callRedstoneChange(world, pos, currentRedstoneLevel, targetRedstoneLevel);

    if (event.getNewCurrent() != targetRedstoneLevel) {
        return;
    }
}
```

keeps the boolean -> redstone level logic in a single place.